tests/utils.py

```diff
@@ -333,17 +335,46 @@ def file_content(path: Path) -> str | bytes | None:
 
 def assert_url_resolvable(url: str):
     """Tests whether a URL is resolvable with retries; raises for failure."""
-    transport = httpx.HTTPTransport(
-        # Retry for connection issues
-        retries=3,
```

Comment:
This retry only applies to connection attempts. `httpx` maintainers decided not to introduced more sophisticated retry handling in to their clients....so third party packages or (custom transports) are necessary to fill this void.